enhancements/installer/assisted-installer-in-cluster.md

```diff
@@ -0,0 +1,539 @@
+parameter so that the ISO’s location isn’t easily guessable.
+* Agent APIs (near-term): Until a point where the agent creates and modifies
+Agent CRs itself, the agent will continue communicating with the service via
+REST APIs. Currently the service embeds the user’s pull secret in the discovery
+ISO which the agent passes in an authentication header.  In this case the
+service can generate and embed some token which it can later validate.
+
+
+#### Hive Integration
+
+Hive has a [ClusterDeployment
+CRD](https://github.com/openshift/hive/blob/master/docs/using-hive.md#clusterdeployment)
+resource that represents a cluster to be created. It includes a "platform"
+section where platform-specific details can be captured. For an agent-based
+workflow, this section will include a new `AgentBareMetal` platform that
+contains such fields as:
+
+* API VIP
+* API VIP DNS name
+* IngressVIP
+
+A new `InstallStrategy` section of the Spec enables the API to describe a way
+of installing a cluster other than the default of using `openshift-install`.
+The new field has an "agent" option that would be common to any Agent-driven
+installation, regardless of platform. Fields include:
+
+* AgentSelector, a label selector for identifying which Agent resources should
+be included in the cluster.
+* ProvisionRequirements, where the API user can specify how many agents to
+expect before beginning installation for each of the control-plane and worker
+roles.
+
+#### Use of metal3
+
+[metal3](https://metal3.io/) will be used to boot the assisted discovery ISO
+for bare-metal deployments. Specifically, the BareMetalHost API will be used
+from a Hub cluster to boot the discovery ISO on hosts that will be used to
+form new clusters.
+
+For this first iteration, it is assumed that the hub cluster itself is using
+the bare metal platform, and thus will have baremetal-operator installed
+and available for use. In the future it may be desirable to make the metal3
+capabilities available on hub clusters that are not using the bare metal
+platform.
+
+#### BareMetalHost can boot live ISOs
+
+A separate [enhancement to
+metal3](https://github.com/metal3-io/metal3-docs/pull/150) proposes a new
+capability in the BareMetalHost API enabling it to boot live ISOs other than
+the Ironic Python Agent. That feature is required so that automation in a
+cluster can boot the discovery ISO on known hardware as the first step toward
+provisioning that hardware.
+
+#### Host Approval
+
+It is important that when an agent checks in, it not be allowed to join a
+cluster until an actor has approved it. From a security standpoint, it is not
+OK for anyone who can access a copy of a discovery ISO and/or access the API
+where agents report themselves to implicitly have the capability to join a
+cluster as a Node.
+
+The Agent CRD will have a field in which to designate that it is approved. If
+the host was booted via the baremetal-operator, approval will be granted
+automatically.
```

Comment:
Who or what will grant the approval automatically?

Reply:
The same automation that caused the host to boot would need to be able to recognize its agent once it comes up and then mark it as approved. I can add (potentially a lot) more detail about that, but it's getting into the area where I'm not sure if it's too much implementation detail. I'll add something similar to the first sentence of this comment, and let me know if more would be useful.

Reply:
It doesn't have to be an exhaustive list of API calls. The point is to make sure we know which actor is doing each action, and show enough of the data flow between steps that it's clear that each actor has the information they need to take those actions.